src/lib/mina_base/parties_logic.ml

```diff
@@ -419,7 +432,7 @@ module Make (Inputs : Inputs_intf) = struct
             ~else_:local_state.ledger
       }
     in
-    let protocol_state_predicate_satisfied =
+    let parties_protocol_state_predicate_satisfied =
```

Comment:
Can we change this to always check, and move the `start_data.protocol_state_predicate` into the fee-payer party?